core/src/test/java/org/apache/iceberg/TestFileScanTaskParser.java

```diff
@@ -27,29 +27,41 @@
 import org.junit.jupiter.params.provider.ValueSource;
 
 public class TestFileScanTaskParser {
+
   @Test
   public void testNullArguments() {
-    Assertions.assertThatThrownBy(() -> FileScanTaskParser.toJson(null))
+    Assertions.assertThatThrownBy(() -> ScanTaskParser.toJson(null))
```

Comment:
I don't think this test class should actually change at all. Since you added a new `ScanTaskParser`, it's best to also add a `TestScanTaskParser`. This test should stay the same, since someone could still use the Parser and the existing behavior of the parser shouldn't change IMO

Reply:
I see your point here. Will add a `TestScanTaskParser` to test the facade part, like task type field and unsupported task type.

You pointed out another issue in this test class. we should test both the old/deprecated public API (via `FileScanTaskParser.toJson/fromJson`) and new `ScanTaskParser` public API.